File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/5. 1436.py
import sys

input = sys.stdin.readline
output = sys.stdout.write

N = int(input())

cnt = 0
result = 666

while True:
  tempRes = result

  con6 = 0
  while tempRes > 0:
    nowNum = tempRes % 10

    if nowNum == 6:
      con6 += 1
    else:
      con6 = 0

    if con6 == 3:
      cnt += 1
      break
    tempRes //= 10
  
  if cnt == N:
    output(f"{result}")
    break

  result += 1

File: 1. 단계별로_풀어보기/10~19/12. 브루트 포스/5. 1436.js
// 입력
const inputArr = require('fs').readFileSync(0).toString().trim().split('\n')

// 입력 파싱
const N = +inputArr[0]

// 몇 번째 666 수인지
let cnt = 0
let result = 666

// 연산
while (true) {
  const numArr = result.toString().split('')

  // 6이 연속으로 몇 번 나왔나
  let con6 = 0

  for (let num of numArr) {
    if (num === '6') {
      con6 += 1
    } //
    else {
      con6 = 0
    }

    if (con6 === 3) {
      cnt += 1
      break
    }
  }

  if (cnt === N) {
    process.stdout.write(`${result}`)
    break
  }

  result += 1
}
